### PR TITLE
[ORPH-14] Enabled publication of project POM and Dockerfile

### LINKED ISSUE #14: Need to publish poms in order to ease dependency needs

### PR DESCRIPTION
**Pull request by peter-mcclonski**
Improves safety of dependency specification between modules.

### CODE REVIEW

File: orphedomos-maven-plugin/src/main/resources/META-INF/plexus/components.xml [hunk 1]
```diff
@@ -7,23 +7,46 @@
...
+					<phase>package</phase>
+					<phase>install</phase>
+					<phase>deploy</phase>
+				</phases>
+				<default-phases>
+					<validate>
+						org.technologybrewery.orphedomos:orphedomos-maven-plugin:verify-docker-environment
+					</validate>
+					<package>
+						org.technologybrewery.orphedomos:orphedomos-maven-plugin:build-docker-image
```
Comment (peter-mcclonski) [inline orphedomos-maven-plugin/src/main/resources/META-INF/plexus/components.xml:28]: I: Considering moving this to the `install` phase, and only retaining the function that sets the additional artifacts in the `package` phase.  It doesn't _really_ matter, but may be slightly more correct.

Comment (d-ryan-ashcraft) [reply orphedomos-maven-plugin/src/main/resources/META-INF/plexus/components.xml:28]: I think ideally, we want the dockerfile builds in `package`.  I just made a similar change in habushu and think we should NOT publish the actual artifact, but instead a placeholder.  The idea is that it prevents any confusion on where the artifact that is being used comes from.  We don't really want Maven to manage the artifact, we want the POM for build efficiency IMO.

Comment (peter-mcclonski) [reply orphedomos-maven-plugin/src/main/resources/META-INF/plexus/components.xml:28]: Right.  The logic here is that we get pom publishing for free from the maven-install-plugin and maven-deploy-plugin, but if we don't specify some sort of primary artifact in addition, it'll throw an error.  I figure the dockerfile is as good an option as anything for that role, though I could also lean towards something like a flat file with a list of all produced tags.

Comment (d-ryan-ashcraft) [reply orphedomos-maven-plugin/src/main/resources/META-INF/plexus/components.xml:28]: Per offline convo, pushed an update that aligns this w/ the pattern we used in habushu.  Pushes a placeholder file:
```
[INFO] --- install:3.1.1:install (default-install) @ orphedomos-test-multimodule-b ---
[INFO] Installing /Users/****/dev/technology-brewery/orphedomos/orphedomos-test/orphedomos-test-multimodule/orphedomos-test-multimodule-b/pom.xml to /Users/****/.m2/repository/org/technologybrewery/orphedomos/orphedomos-test-multimodule-b/0.8.0-SNAPSHOT/orphedomos-test-multimodule-b-0.8.0-SNAPSHOT.pom
[INFO] Installing /Users/****/dev/technology-brewery/orphedomos/orphedomos-test/orphedomos-test-multimodule/orphedomos-test-multimodule-b/target/orphedomos.placeholder.txt to /Users/****/.m2/repository/org/technologybrewery/orphedomos/orphedomos-test-multimodule-b/0.8.0-SNAPSHOT/orphedomos-test-multimodule-b-0.8.0-SNAPSHOT.orphedomos.placeholder.txt
```

**Review by d-ryan-ashcraft — APPROVED** [review]

**Pull request merged**